src/Database/LSMTree/Internal/WriteBufferBlobs.hs

```diff
@@ -0,0 +1,245 @@
+{-# OPTIONS_GHC -Wno-partial-fields #-}
+
+-- | An on-disk store for blobs for the write buffer.
+--
+-- For table inserts with blobs, the blob get written out immediately to a
+-- file, while the rest of the 'Entry' goes into the 'WriteBuffer'. The
+-- 'WriteBufferBlobs' manages the storage of the blobs.
+--
+-- A single write buffer blob file can be shared between multiple table handles.
+-- As a consequence, the lifetime of the 'WriteBufferBlobs' must be managed
+-- using 'newWriteBufferBlobs','addReference' and 'removeReference'. A fresh
+-- 'WriteBufferBlobs' has a reference count of 1, and so must be matched by
+-- 'removeReference'. When a table handle is duplicated, the new handle needs
+-- its own reference, so use 'addReference' upon duplication.
+--
+-- Blobs are copied from the write buffer blob file when the write buffer is
```

Comment:
Yes, I realised as I was implementing it that it wasn't possible to reuse the write buffer's blob file to become the run's blob file, as was our initial intuition. The reason being that the need to share the blob file between multiple table handles with O(1) duplication cost, means the blob file content isn't stable at the point at which we want to flush it.

One might be able to avoid the copy, but it'd need a change to the format of snapshot blob files too.